src/Disciplina/Core/Types.hs

```diff
@@ -62,15 +67,59 @@ newtype CourseId = CourseId
     { getCourseId :: Word32
     } deriving (Eq, Ord, Show, Num)
 
+-- | Assignment can be either regular of final
+data AssignmentType = Regular | CourseFinal
+    deriving (Eq, Show, Generic)
+
+-- | Assignment doesn't contain actual assignment contents - only hash of them.
+data Assignment = Assignment
+    { aCourseId   :: !CourseId
+    -- ^ Course this assignement belongs to
+    , aType       :: !AssignmentType
+    -- ^ Assignment type
+    , aAssignment :: !Text
+    -- ^ Description of assignment
+    } deriving (Eq, Show, Generic)
```

Comment:
As I already said, it probably makes sense to store contents' hash for assignment and submission separately. This way we also might get rid of `AssignmentType` inside `Assignment` (the same for `SubmissionType`), because it can be encoded into contents' hash. I'll think about it a little bit and write down the necessary spec in issue's comments.